apps/etl-func/src/adapters/functions/messages-ingestion.ts

```diff
@@ -0,0 +1,57 @@
+import { CosmosDBHandler } from "@azure/functions";
+import { pino } from "pino";
+
+import {
+  Message,
+  MessageEvent,
+  MessageMetadata,
+  messageMetadataSchema,
+} from "../../domain/message.js";
+import { MessageAdapter, getMessageEventFromMessage } from "../message.js";
+import { EventHubEventProducer } from "../message-event.js";
+import PDVTokenizerClient from "../pdv-tokenizer/pdv-tokenizer-client.js";
+
+const logger = pino({
+  level: process.env.NODE_ENV === "production" ? "error" : "debug",
+});
+
+const messagesIngestion =
+  (
+    messageAdapter: MessageAdapter,
+    PDVTokenizer: PDVTokenizerClient,
+    producer: EventHubEventProducer<MessageEvent>,
+  ): CosmosDBHandler =>
+  async (documents: unknown[]) => {
+    //Avoiding all documents different from MessageMetadata schema
+    const documentsMetadata: MessageMetadata[] = documents.filter(
+      (item): item is MessageMetadata =>
+        messageMetadataSchema.safeParse(item).success,
+    );
+
+    try {
+      //Retrieving the message contents for each message metadata
+      const messagesContet = (
+        await Promise.all(
+          documentsMetadata.map((messageMetadata) =>
+            messageAdapter.getMessageByMetadata(messageMetadata),
+          ),
+        )
+      ).filter(
+        (item: Message | undefined): item is Message => item !== undefined,
```

Comment:
Ditto, typescript can infer the type automatically here
```suggestion
        (item): item is Message => item !== undefined,
```